src/estimators/bloom_filters.py

```diff
@@ -287,6 +316,21 @@ def _estimate_cardinality_uniform(cls, sketch):
     m = len(sketch.sketch)
     return - m * math.log(1 - x / m)
 
+  @classmethod
+  def _estimate_cardinality_geo(cls, sketch):
+    """Estimate cardinality of an Geometric Bloom Filter.
+
+    Args:
+      sketch: An GeometricBloomFilter. It should be unnoised or obtained
+        after denoising.
+    Returns:
+      The estimated cardinality of the BF.
+    """
+    z = np.sum(sketch.sketch)
+    k = float(sketch.num_hashes())
+    m = float(sketch.max_size())
+    return int(abs(math.log(z/m) / (k * math.log(1 - 1 / m))))
```

Comment:
Would you please double check this estimate which doesn't seem right?

For example, if the sketch is almost full, i.e. z -> m, the numerator -> 0. Yet the denominator is irrelevant of z, so the estimate -> 0.

I suspect that the denominator also depends on z, maybe k * math.log(1 - z / m)?

Reply:
Should the unique estimator depends on the geometric probability=0.008?

Reply:
Used _estimate_cardinality_any, since we just need to solve N for the equation of E(N) = X